### ISSUE TITLE
UI is disabled even with ui_config.enabled=true

### ISSUE THREAD

**Issue by fenfir** [hashicorp/consul-k8s#1201]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request. Searching for pre-existing feature requests helps us consolidate datapoints for identical requirements into a single place, thank you!
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request.
* If you are interested in working on this issue or have submitted a pull request, please leave a comment.

---

### Overview of the Issue

After installing the helm chart with the default values, the UI is unavailable. When I got to localhost:8500 I just get a page with 

```
Consul Agent: UI disabled. To enable, set ui_config.enabled=true in the agent configuration and restart.
```

### Reproduction Steps

Install using helm with the default values

```
helm template consul consul --version 0.43.0 --namespace dc3-monitoring --repo https://helm.releases.hashicorp.com --values values.yaml
```

the values.yaml used above was generated by
```
helm show values consul --version 0.43.0 --repo https://helm.releases.hashicorp.com  > values.yaml
```

port foward the UI 
```
kubectl port-forward service/consul-consul-server 8500:8500
```

go to http://localhost:8500

### Logs

There are no relavant logs to show.

### Expected behavior

To see the Consul UI at http://localhost:8500

### Environment details
chart version 0.43.0

The configmap contains the ui_config.enabled=true

```
apiVersion: v1
data:
  central-config.json: |-
    {
      "enable_central_service_config": true
    }
  extra-from-values.json: |-
    {
      "log_level": "DEBUG"
    }
  server.json: |
    {
      "auto_reload_config": true,
      "bind_addr": "0.0.0.0",
      "bootstrap_expect": 3,
      "client_addr": "0.0.0.0",
      "connect": {
        "enabled": true
      },
      "datacenter": "dc1",
      "data_dir": "/consul/data",
      "domain": "consul",
      "ports": {
        "serf_lan": 8301
      },
      "recursors": [],
      "retry_join": ["consul-consul-server.dc3-monitoring.svc:8301"],
      "server": true
    }
  ui-config.json: |-
    {
      "ui_config": {
        "enabled": true
      }
    }
kind: ConfigMap
```

**Comment by t-eckert** [comment]
Hi @fenfir,

Thank you for reaching out. Given the information here, you are correct that the configuration should be giving you access to a UI. Would you please try following the instructions [here](https://www.consul.io/docs/k8s/installation/install) and see if you get the same result?

**Comment by fenfir** [comment]
Using the helm install command from there the ui isn't enabled at all.

```
helm install consul hashicorp/consul --set global.name=consul --create-namespace --namespace consul --version 0.43.0
kubectl port-forward service/consul-server --namespace consul 8500:8500
```
The ui_config section exists in the configmap
```
    ui-config.json: |-
      {
        "ui_config": {
          "enabled": true
        }
      }
``` 

I still see this in the browser
```
Consul Agent: UI disabled. To enable, set ui_config.enabled=true in the agent configuration and restart.
```

**Comment by fenfir** [comment]
Is there any update on this? I am still unable to access  the UI.

**Comment by lkysow** [comment]
Can you share your `values.yaml`? I'm really confused how this is happening!

**Comment by lkysow** [comment]
Oh sorry, I see you used `helm show values consul --version 0.43.0 --repo https://helm.releases.hashicorp.com `

**Comment by lkysow** [comment]
A couple things:

1. I can't reproduce this. Maybe you have a `consul agent -dev` running locally that's stealing that port? What if you do a port-forward using a different port? `kubectl port-forward service/consul-consul-server 9500:8500` and then try localhost:9500?
2. Don't create your values.yaml in the way that you have. Instead, only create a values.yaml that overrides the settings you need to override. If you do it like you've said then using future versions of the chart will be very hard because we don't expect users to do it like that.

**Comment by fenfir** [comment]
Even with out creating a values file I still get the same thing. If you look at my previous reply even just with helm install it doesn't work.

**Comment by fenfir** [comment]
I'm closing this. I tired the same commands on a different cluster and it worked.

**Comment by lekaihuai** [comment]
> I'm closing this. I tired the same commands on a different cluster and it worked.

i'm also facing this problem, would you tell me what's diffrence between your clusters?

**Comment by litch** [comment]
I just got bitten by this as well.  Not sure why.  =/

**Comment by aparedero** [comment]
Quite late to reply but today I found this same issue while deploying Consul with Helm and I got it fixed.

# TL;DR

Consul Helm chart has value `ui.enabled: "-". Explicitly set ui block in your consul-values.yaml file with `ui.enabled = true`

```yaml
[...]
ui:
  enabled: true
  service:
    enabled: true
    type: null
    port:
      http: 80
      https: 443
[...]
```

and Edit service `*-consul-ui` to expose via NodePort or LoadBalancer

# Explanation

I had a basic consul-values.yaml file which doesnt have references to the ui itself:

```yaml
global:
  datacenter: eu-west-1-consul

client:
  enabled: true

server:
  replicas: 1
  bootstrapExpect: 1
  disruptionBudget:
    maxUnavailable: 0
```
So checking what are the default Helm chart values clearing comments:
```bash
 helm show values hashicorp/consul | grep -v "#"
```

ui block appears with following configuration:

```yaml
ui:
  enabled: "-"
  service:
    enabled: true
    type: null
    port:
      http: 80
      https: 443
    nodePort:
      http: null
      https: null
    annotations: null
    additionalSpec: null
```

Copy that config and modify `enabled "-"`  with `enabled: true`.

Upgrade your deployment with `helm upgrade <your release name> hashicorp/consul -f consul-values.yaml`.
Then check your services, you should see a service called consul-consul-ui

```
NAME                             TYPE           CLUSTER-IP      EXTERNAL-IP   PORT(S)                                                                            AGE
consul-consul-connect-injector   ClusterIP      10.43.118.183   <none>        443/TCP                                                                            85m
consul-consul-dns                ClusterIP      10.43.152.143   <none>        53/TCP,53/UDP                                                                      85m
consul-consul-server             ClusterIP      None            <none>        8500/TCP,8502/TCP,8301/TCP,8301/UDP,8302/TCP,8302/UDP,8300/TCP,8600/TCP,8600/UDP   85m
consul-consul-ui                 ClusterIP      10.43.81.241    <none>        80/TCP                                                                             85m
```

Modify the service consul-consul-ui from type: ClusterIP to type: NodePort or LoadBalancer. In my case, NodePort
Check what's the dynamic port assigned (or set a custom port on NodePort pool (30000-32767)
```
consul-consul-ui                 NodePort       10.43.81.241    <none>        80:31330/TCP 
```
